src/Surface.zig

```diff
@@ -3275,7 +3275,21 @@ fn processLinks(self: *Surface, pos: apprt.CursorPos) !bool {
                 .trim = false,
             });
             defer self.alloc.free(str);
-            try internal_os.open(self.alloc, .unknown, str);
+
+            // Handle paths starting with ~
+            if (str.len > 0 and str[0] == '~') {
+                const home = std.posix.getenv("HOME") orelse "";
```

Comment:
We should be using `internal_os.expandHome` instead of repeating this logic everywhere.